### PR TITLE
Add support for expression compilation when debugging integration tests

### LINKED ISSUE #79439: Support expresssion evaluation when debugging Integration Tests with the Flutter Tool
Following up on #66264. Users today with `flutter test` are able to evaluate arbitrary expressions with an IDE while debugging tests. This is not supported when doing so for Integration Tests though. The following error is printed when trying to do so:

```
_compileExpression: No compilation service available; cannot evaluate from source.
```

### PR DESCRIPTION
**Pull request by DanTup**
Fixes #79439.

*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by DanTup** [comment]
@jonahwilliams @christopherfujino I don't know if this is the right approach, but it seems to resolve the issue.

I think the failure is unrelated though (it's in `flutter_build_with_compilation_error_test` and expects a Dart compile error but gets an error that ends with "iOS 16.0 is not installed. To use with Xcode, first download and install the platform").

File: packages/flutter_tools/test/integration.shard/expression_evaluation_test.dart [hunk 1]
```diff
@@ -189,4 +221,5 @@ void expectValue(ObjRef result, String message) {
 void main() {
   batch1();
   batch2();
+  batch3();
```
Comment (christopherfujino) [inline packages/flutter_tools/test/integration.shard/expression_evaluation_test.dart:224]: you have any idea why we batch them like this?

**Review by christopherfujino — APPROVED** [review]
LGTM, thanks for wiring this up!

**Comment by christopherfujino** [comment]
Your test failure is an infra issue, re-running

File: packages/flutter_tools/test/integration.shard/expression_evaluation_test.dart (hunk 1 above)
Comment (DanTup) [reply packages/flutter_tools/test/integration.shard/expression_evaluation_test.dart:224]: Not for certain.. They were added in https://github.com/flutter/flutter/pull/41880/files

I figured maybe it was for some consistency and scoping. Those in each batch use a different `TestProject` and each "batch" has a similar `initProject`, `cleanProject` function. I'm not sure I see a reason why they wouldn't have just been inside `group()`s though. I can try changing them to be that way though (either in this PR or another)?

Comment (DanTup) [reply packages/flutter_tools/test/integration.shard/expression_evaluation_test.dart:224]: Gonna merge this, but let me know if you think I should try changing these to `group`s in a future CL. Base on other things in the change above, maybe this was intended to provide some sharding, but as far as I can tell nobody is calling these `batch()` functions individually.

**Pull request merged**